src/ceph-volume/ceph_volume/devices/lvm/listing.py

```diff
@@ -127,149 +95,88 @@ def list(self, args):
-
-        if lv:
-            try:
-                _id = lv.tags['ceph.osd_id']
-            except KeyError:
-                logger.warning('device is not part of ceph: %s', device)
-                return report
-
-            report.setdefault(_id, [])
+            osd_id = lv.tags['ceph.osd_id']
+            report.setdefault(osd_id, [])
             lv_report = lv.as_dict()
-            lv_report['devices'] = self.match_devices(lv.lv_uuid)
-            report[_id].append(lv_report)
 
-        else:
-            # this has to be a journal/wal/db device (not a logical volume) so try
-            # to find the PARTUUID that should be stored in the OSD logical
-            # volume
-            for device_type in ['journal', 'block', 'wal', 'db']:
-                device_tag_name = 'ceph.%s_device' % device_type
-                device_tag_uuid = 'ceph.%s_uuid' % device_type
-                associated_lv = lvs.get(lv_tags={device_tag_name: device})
-                if associated_lv:
-                    _id = associated_lv.tags['ceph.osd_id']
-                    uuid = associated_lv.tags[device_tag_uuid]
-
-                    report.setdefault(_id, [])
-                    report[_id].append(
-                        {
-                            'tags': {'PARTUUID': uuid},
-                            'type': device_type,
-                            'path': device,
-                        }
-                    )
+            dev_type = lv.tags.get('ceph.type', None)
+            if dev_type != 'journal' or (dev_type == 'journal' and
+               full_report):
+                pvs = api.get_pvs(filters={'lv_uuid': lv.lv_uuid})
+                lv_report['devices'] = [pv.name for pv in pvs] if pvs else []
+                report[osd_id].append(lv_report)
+
+            if arg_is_vg or dev_type in ['journal', 'wal']:
+                create_report_for_nonlv_device()
+
         return report
 
-    def full_report(self, lvs=None):
+    def full_report(self):
         """
-        Generate a report for all the logical volumes and associated devices
-        that have been previously prepared by Ceph
+        Create a report of all Ceph LVs. Returns '{}' to denote failure.
         """
-        if lvs is None:
-            lvs = api.Volumes()
-        report = {}
+        return self.create_report(api.get_lvs())
 
-        for lv in lvs:
-            try:
-                _id = lv.tags['ceph.osd_id']
-            except KeyError:
-                # only consider ceph-based logical volumes, everything else
-                # will get ignored
-                continue
+    def single_report(self, device):
+        """
+        Generate a report for a single device. This can be either a logical
+        volume in the form of vg/lv or a device with an absolute path like
+        /dev/sda1 or /dev/sda. Returns '{}' to denote failure.
+        """
+        lv = api.get_first_lv(filters={'lv_path': device})
```

Comment:
I think there's an issue with this call, shouldn't `filters` be converted using `convert_filters_to_str()`?

Reply:
@rishabh-d-dave @jan--f

Reply:
`get_first_lv` -> `get_lvs` -> `make_filters_lvmcmd_ready` -> `convert_filters_to_str`